feat: add mcp page to sdk website (#702)
kubeflow/sdk: docs/source/mcp-server/tools.rst

Tools
=====

> The Kubeflow MCP Server exposes 23 tools organized by workflow phase.

Reviewer: Maintenance note: the literal `23` is hard-coded in four places (here, `:53`, `:63`, and `mcp-server/index.rst:27`). One tool added upstream makes all four wrong, and nobody watching this repo gets a signal. Consider a non-numeric phrasing ("tools, grouped by workflow phase") and letting the catalog table carry the detail. Same rot risk applies to the env-var table, the CLI flag list, and the version-compatibility table — deep-linking to the upstream README for those would cut a lot of future drift.

Tool Catalog
------------

.. list-table::
   :header-rows: 1
   :widths: 15 40 45

   * - Phase
     - Tools
     - Description
   * - Planning
     - ``pre_flight``, ``check_compatibility``, ``get_cluster_resources``, ``estimate_resources``
     - Environment validation and resource estimation
   * - Discovery
     - ``list_training_jobs``, ``get_training_job``, ``list_runtimes``, ``get_runtime``
     - Browse jobs and available runtimes
   * - Training
     - ``fine_tune``, ``run_custom_training``, ``run_container_training``
     - Submit LoRA/QLoRA fine-tuning, custom scripts, or container jobs
   * - Monitoring
     - ``get_training_logs``, ``get_training_events``, ``wait_for_training``
     - Track progress and debug failures
   * - Lifecycle
     - ``delete_training_job``, ``update_training_job``
     - Manage existing jobs (ownership-guarded)
   * - Platform
     - ``inspect_crd``, ``inspect_controller``, ``patch_runtime``, ``create_runtime``, ``delete_runtime``
     - Cluster inspection and runtime management
   * - Health
     - ``health_check``, ``get_server_logs``
     - Server diagnostics

> Every mutating tool (``fine_tune``, ``run_custom_training``, ``run_container_training``,
> ``delete_training_job``, ``update_training_job``, ``patch_runtime``, ``create_runtime``,
> ``delete_runtime``) requires ``confirmed=True`` — the first call always returns a preview.

Reviewer: `update_training_job` has no `confirmed` parameter — it patches the TrainJob immediately ([`trainer/api/lifecycle.py:120`](https://github.com/kubeflow/mcp-server/blob/main/kubeflow_mcp/trainer/api/lifecycle.py#L120)). Only 7 tools take `confirmed`. An agent told "every mutating tool previews first" will suspend a job it meant to dry-run.

```suggestion
Most mutating tools (``fine_tune``, ``run_custom_training``, ``run_container_training``,
``delete_training_job``, ``patch_runtime``, ``create_runtime``, ``delete_runtime``) require
``confirmed=True`` — the first call returns a preview. ``update_training_job`` is the
exception: it applies the suspend/resume patch immediately.
```

The same qualification is needed at `mcp-server/index.rst:24`, `:117`, `:131` and `configuration.rst:113`. Upstream's `README.md`/`ROADMAP.md` make the same claim, so this is worth an issue on kubeflow/mcp-server too.

Reviewer 2: This issue is tracked here and we are expecting this fix will land under 0.1.0 release.. so we should be good here : https://github.com/kubeflow/mcp-server/issues/121

Tool Discovery Modes
---------------------

.. list-table::
   :header-rows: 1
   :widths: 20 15 65

   * - Mode
     - Tools exposed
     - Description
   * - ``full``
     - 23

>      - All tools registered directly (default)

Reviewer: Mode defaults to `full`, but persona defaults to `readonly`, which allows only 12 tools ([`core/policy.py:74-89`](https://github.com/kubeflow/mcp-server/blob/main/kubeflow_mcp/core/policy.py#L74)) and `create_server` intersects the two before registering. A default `kubeflow-mcp serve` exposes 12 tools, not 23 — only `platform-admin` gets all 23.

```suggestion
   * - ``full``
     - up to 23
     - All persona-allowed tools registered directly (default mode; the default
       ``readonly`` persona exposes 12 of the 23)
```

>    * - ``progressive``
>      - ~3 meta-tools
>      - Hierarchical discovery (~85 tokens); agents drill down by phase
>    * - ``semantic``
>      - ~2 meta-tools
>      - Embedding-search discovery (~69 tokens); agents query by intent

Reviewer: These counts are exact, not approximate — `PROGRESSIVE_TOOLS` has 3 entries and `SEMANTIC_TOOLS` has 2 (`core/dynamic_tools.py:219,381`). The `~` belongs on the token figures only.

```suggestion
   * - ``progressive``
     - 3 meta-tools
     - Hierarchical discovery (~85 tokens); agents drill down by phase
   * - ``semantic``
     - 2 meta-tools
     - Embedding-search discovery (~69 tokens); agents query by intent
```

``progressive`` and ``semantic`` modes significantly reduce token consumption for agent
workflows compared to registering all 23 tools directly.

Requirements
------------

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 20 20

   * - MCP Server
     - Kubeflow Trainer
     - Kubeflow SDK
     - Python
     - Kubernetes
   * - 0.1.x
     - >= 2.2.0

>      - >= 0.4.0

Reviewer: `pyproject.toml:22` pins the SDK exactly (`kubeflow==0.4.0`, same for the `hub`/`spark`/`all` extras), so `>= 0.4.0` promises compatibility the package refuses — installing alongside any other SDK version is a resolver conflict.

```suggestion
     - == 0.4.0
```

     - 3.10 - 3.12
     - >= 1.27

Health and Readiness
---------------------

Container and Kubernetes probes are available without MCP authentication:

.. code-block:: text

   GET /health  # liveness: the server process is accepting HTTP requests
   GET /ready   # readiness: configured clients imported and packaged resources loaded

> ``/ready`` returns 200 only when both ``clients_ready`` and ``resources_ready`` are true. It

Reviewer: `clients_ready` and `resources_ready` are Python locals in `core/server.py`, not response fields — the actual bodies are `{"status": "ready"}` / `{"status": "not_ready"}` (`core/http_edge.py:27-33`). Literal markup invites a reader to `jq .clients_ready` and get `null`.

```suggestion
``/ready`` returns ``{"status": "ready"}`` with HTTP 200 only when the configured clients
imported successfully **and** all packaged resources loaded; otherwise it returns
``{"status": "not_ready"}`` with HTTP 503. It
```
